build/build-rustc-cg-gcc.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+
+## $1 : version, currently rustc_cg_gcc does not have any and only uses master branch.
+## $2 : destination: a directory or S3 path (eg. s3://...)
+
+set -ex
+
+ROOT=$PWD
+VERSION=$1
+
+if [[ "${VERSION}" != "master" ]]; then
+    echo "Only support building master"
+    exit 1
+fi
+
+FULLNAME=rustc-cg-gcc-${VERSION}-$(date +%Y%m%d)
+
+OUTPUT=${ROOT}/${FULLNAME}.tar.xz
+S3OUTPUT=
+if [[ $2 =~ ^s3:// ]]; then
+    S3OUTPUT=$2
+else
+    if [[ -d "${2}" ]]; then
+        OUTPUT=$2/${FULLNAME}.tar.xz
+    else
+        OUTPUT=${2-$OUTPUT}
+    fi
+fi
+
+## From now, no unset variable
+set -u
+
+OUTPUT=$(realpath "${OUTPUT}")
+
+rm -rf  build-rustc-cg-gcc
+mkdir -p build-rustc-cg-gcc
+
+pushd build-rustc-cg-gcc
+PREFIX=$(pwd)/gcc-install
+
+export CARGO_HOME=$PWD/rustup
+export RUSTUP_HOME=$PWD/rustup
+
+export PATH=$RUSTUP_HOME/bin:$PATH
+
+## Download rustc_cg_gcc
+git clone --depth 1 https://github.com/antoyo/rustc_codegen_gcc.git
+
+## Download rustup and install it in a local dir
+## Installs :
+## - minimal profile
+## - the required build-deps from RUSTUP_COMP_BUILD_DEPS (-c *)
+## - version taken from rust-toolchain file
+
+RUSTUP_COMP_BUILD_DEPS=(rust-src rustc-dev llvm-tools-preview)
+
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --profile minimal \
```

Comment:
nitpick: can we line wrap this at the pipe so the sh continuation is not off in space?

```
curl .... |
    sh ...
```

Reply:
done